confgenerator/logging_processors.go

```diff
@@ -229,3 +231,30 @@ var LegacyBuiltinProcessors = map[string]LoggingProcessor{
 		},
 	},
 }
+
+// A LoggingProcessorExcludeLogs filters out logs according to a pattern.
+type LoggingProcessorExcludeLogs struct {
+	ConfigComponent `yaml:",inline"`
+	MatchesAny      []string `yaml:"matches_any" validate:"required,filter_list"`
+}
+
+func (r LoggingProcessorExcludeLogs) Type() string {
+	return "exclude_logs"
+}
+
+func (p LoggingProcessorExcludeLogs) Components(tag, uid string) []fluentbit.Component {
+	filters := make([]filter.Filter, 0, len(p.MatchesAny))
+	for _, condition := range p.MatchesAny {
```

Comment:
Future optimization, perhaps, but leave a TODO here about handling these with a single `grep` filter?

Reply:
`filter.AllComponents` already emits only a single `grep` filter for multiple `Filter`s. I do think the filter code needs a small refactor though -- I realized `filter.Components` doesn't include the `nest`/`grep`/`lift` anymore which should be fixed.

Reply:
(Or were you referring to the optimization of handling multiple consecutive `exclude_logs` processors with a single `grep` filter?)

Reply:
Added a comment.